src/rpc/ngHandlers/NoRippleCheck.cpp

```diff
@@ -0,0 +1,220 @@
+            if (limit != 0 && ownedItem.getType() == ripple::ltRIPPLE_STATE)
+            {
+                bool const bLow = accountID ==
+                    ownedItem.getFieldAmount(ripple::sfLowLimit).getIssuer();
+
+                bool const bNoRipple = ownedItem.getFieldU32(ripple::sfFlags) &
+                    (bLow ? ripple::lsfLowNoRipple : ripple::lsfHighNoRipple);
+
+                std::string problem;
+                bool needFix = false;
+                if (bNoRipple & input.roleGateway)
+                {
+                    problem = "You should clear the no ripple flag on your ";
+                    needFix = true;
+                }
+                else if (!bNoRipple & !input.roleGateway)
+                {
+                    problem =
+                        "You should probably set the no ripple flag on "
+                        "your ";
+                    needFix = true;
+                }
+                if (needFix)
+                {
+                    limit--;
+                    ripple::AccountID peer =
+                        ownedItem
+                            .getFieldAmount(
+                                bLow ? ripple::sfHighLimit : ripple::sfLowLimit)
+                            .getIssuer();
+                    ripple::STAmount peerLimit = ownedItem.getFieldAmount(
+                        bLow ? ripple::sfHighLimit : ripple::sfLowLimit);
+                    problem += fmt::format(
+                        "{} line to {}",
+                        to_string(peerLimit.getCurrency()),
+                        to_string(peerLimit.getIssuer()));
+                    output.problems.emplace_back(problem);
+                    if (input.transactions)
+                    {
+                        ripple::STAmount limitAmount(ownedItem.getFieldAmount(
+                            bLow ? ripple::sfLowLimit : ripple::sfHighLimit));
+                        limitAmount.setIssuer(peer);
+                        auto tx = getBaseTx(*accountID, accountSeq++);
+                        tx[JS(TransactionType)] = "TrustSet";
+                        tx[JS(LimitAmount)] = RPC::toBoostJson(
+                            limitAmount.getJson(ripple::JsonOptions::none));
+                        tx[JS(Flags)] = bNoRipple ? ripple::tfClearNoRipple
+                                                  : ripple::tfSetNoRipple;
+                        output.transactions->push_back(tx);
+                    }
+                }
+            }
+            return true;
+        });
+
+    output.ledgerIndex = lgrInfo.seq;
+    output.ledgerHash = ripple::strHex(lgrInfo.hash);
+    return output;
+}
+
+NoRippleCheckHandler::Input
+tag_invoke(
+    boost::json::value_to_tag<NoRippleCheckHandler::Input>,
+    boost::json::value const& jv)
+{
+    auto const& jsonObject = jv.as_object();
+    NoRippleCheckHandler::Input input;
+    input.account = jv.at(JS(account)).as_string().c_str();
```

Comment:
I wonder, why not use jsonObject here too?

Reply:
json object and value both have at() function. jsonObject here is mainly for contains() function.